accepted/2024/vulnerabilities-in-packages.config-restore.md

```diff
@@ -0,0 +1,93 @@
+
+## Motivation
+
+See [vulnerabilities in PackageReference restore motivation](../2022/vulnerabilities-in-restore.md#motivation).
+
+## Explanation
+
+### Functional explanation
+
+Packages.config restore combines all packages regardless of project. Instead of raising warnings and errors on a per project level, at restore time, they're raised on a per package level.
+
+The configuration knobs for packages.config restore auditing functionality will be NuGet.config based.
+In particular, there will be 2 new configuration keys:
+
+| Key | Acceptable values | Description | Default |
+|-----|-------------------|-------------|---------|
+| auditForPackagesConfig | enable, disable | Enables or disables NuGet Audit for packages config projects | If not specified, the default will be `enable` |
+| auditLevelForPackagesConfig | Critical, high, moderate, low | Configures the default audit level for NuGet audit for packages config projects |  If not specified, the default will be `low` |
+
+The audit functionality for packages.config restore will be enabled by default.
+To disable it, one can specify a property in the config section of the configuration file.
+
+```xml
+<configuration>
+    <config>
+        <add key="audit" value="enable" />
+        <add key="auditLevel" value="low" />
+    </config>
+</configuration>
+```
+
+Whenever a vulnerability for a package is discovered, a warning will be raised indicating the severity and advisory url.
+Note that warnings as errors and no warn are not support in packages.config projects, so these warnings will not fail the build.
+
+> Package 'Contoso.Service.APIs' 1.0.3 has a known critical severity vulnerability, https://github.com/advisories/GHSA-1234-5678-9012.
+
+### Technical explanation
+
+An [AuditUtility](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.PackageManagement/AuditUtility.cs) already exists for packages.config projects, written along similar lines as the AuditUtility for PackageReference.
+
+When we run restore for packages.config, the following metrics will be considered:
+
+- Was Audit enabled
+- Was Audit Run
+- Reason why audit was not run. (No package downloads for example)
+- Sev0Matches
+- Sev1Matches
+- Sev2Matches
+- Sev3Matches
+- InvalidSevMatches
+- List of package ids with advisories
+- Severity
+- DownloadDurationSeconds
+- CheckPackagesDurationSeconds
+- SourcesWithVulnerabilityData
+
+All of these metrics are going to the added to the root vs/nuget/restoreinformation event.
+
+## Drawbacks
+
+- Vulnerability checking comes with a performance hit.
+
+## Rationale and alternatives
+
+- Enable vulnerability checking on demand.
+
+## Prior Art
+
+- [Vulnerability reporting in PackageReference restore](../2022/vulnerabilities-in-restore.md)
+
+## Unresolved Questions
+
+- Should we use MSBuild properties instead?
+- Should the NuGet.config configuration *affect* the PackageReference defaults as well?
```

Comment:
Not sure I like the idea of nuget.config config being introduced here nor affecting PR.

Reply:
I recommend reading: https://github.com/NuGet/Home/blob/dev-nkolev92-vulnerabilityCheckingInPCRestore/accepted/2024/vulnerabilities-in-packages.config-restore.md#msbuild-properties-vs-nugetconfig-for-packagesconfig-restore-configuration. 

It covers the in depth thought process. 
Let me know what you think.